File: apple-and-orange/apple-and-orange.js
// https://www.hackerrank.com/challenges/apple-and-orange/problem

const countApplesAndOranges = (
  houseStart,
  houseEnd,
  aCoord,
  oCoord,
  aDistances,
  oDistances
) => {
  const hitTotals = [
    { distances: aDistances, coord: aCoord },
    { distances: oDistances, coord: oCoord }
  ].map(({ distances, coord }) =>
    distances.reduce((total, dist) => {
      const fruitCoord = coord + dist;
      const didHit = fruitCoord >= houseStart && fruitCoord <= houseEnd;
      return didHit ? ++total : total;
    }, 0)
  );

  hitTotals.forEach(hitTotal => console.log(hitTotal));
};

File: apple-and-orange/apple-and-orange.clj
(defn get-hit-total [distances tree-coord house-start house-end]
  (reduce
   (fn [total d]
     (let [fruit-coord (+ tree-coord d)
           did-hit? (and
                     (>= fruit-coord house-start)
                     (<= fruit-coord house-end))]
       (if did-hit? (inc total) total)))
   0
   distances))

(defn countApplesAndOranges [house-start house-end a-coord o-coord a-distances o-distances]
  (let [a-hit-total (get-hit-total a-distances a-coord house-start house-end)
        o-hit-total (get-hit-total o-distances o-coord house-start house-end)]
    (doseq [hit-total [a-hit-total o-hit-total]] (prn hit-total))))